Remove warn! for pool already known

tasleson opened the pull request:
This message gets printed often and is expected with us monitoring
both 'add' and 'change' udev events.  Add additional check for the
more hideous issue of having a device which thinks it belongs in a
pool, but the existing active pool has no knowledge of it.

Removes the `warn` but does not fully address what we should do for https://github.com/stratis-storage/stratisd/issues/748

Signed-off-by: Tony Asleson <tasleson@redhat.com>

mulkieran commented:
Please see https://github.com/stratis-storage/stratisd/pull/1163. There is an alternative approach, which involves hoisting the ```get_by_uuid``` method up and making it the primary if condition, instead of ```contains_uuid```. But because of borrowing restrictions, it needs to change other things more aggressively, so I figured this was better.

tasleson commented:
@mulkieran AFAIK, `pool.get_blockdev` returns a `BlockDev`, not a `StratBlockDev`.

mulkieran commented:
@tasleson Ah, I see. You're probably better off doing something like this: https://github.com/stratis-storage/stratisd/pull/1163.

tasleson commented:
@mulkieran Would it be cleaner to add a method `dev` to `BlockDev` instead?

tasleson commented:
Replace by: https://github.com/stratis-storage/stratisd/pull/1171

@mulkieran I picked up your changes and squashed.  I did find an issue with the error message:

Changed

```rust
                        if device != *block_dev.device() {
                            error!(
                                "we have a block device with the same uuid as one already in the pool, but the one in the pool has device number {:}, while the one just found has device number {:}",
                                device,
                                block_dev.device()
                            );
```
To
```rust
                       if device != *block_dev.device() {
                            error!(
                                "we have a block device with the same uuid as one already in the \
                                 pool, but the one in the pool has device number {:}, \
                                 while the one just found has device number {:}",
                                block_dev.device(),
                                device,
                            );
                        }
```
The ordering of the variables was incorrect,  `device` contains the just found device and `block_dev.device()` is the device that is in the pool.
